gulpfile.babel.js

```diff
@@ -47,6 +49,16 @@ function createBundle({entries, output, destination},
     b.transform(envify({
         NODE_ENV: production ? 'production' : 'development'
     }), {global: true})
+
+    if(cssOutput) {
+        b.plugin('css-modulesify', {
+            output: path.join(destination, cssOutput),
+            postcssBefore: [
+                require('postcss-cssnext')
```

Comment:
I would import this right at the top rather than require it here. Also 'css-modulesify' could be imported, rather than passed as a string, for clarity and consistency.
Besides that this looks like a neat approach. Runs fine.